### PR TITLE
WEB-78: Backend cleanup

### PR DESCRIPTION
**Pull request by NWCalvank**
Changes informed by the most-recent use of this boilerplate.

**Note**
The FE will need to be updated to handle the new API endpoint(s), but the relevant code has not yet been merged into `master`.

### CODE REVIEW

File: server/test/test_foo.py [hunk 1]
```diff
@@ -1,70 +1,65 @@
...
-        self.assertEqual(data['example']['string_field'], test_string)
+        self.assertEqual(data['string_field'], test_string)
 
-    def test_put_example(self):
-        create_example_string('example_string_1')
+    def test_put_foo(self):
+        create_foo_string('foo_string_1')
         with self.client:
-            response = self.client.put('/api/example_endpoint',
+            response = self.client.put('/api/foo/1',
```
Comment (fwkendall) [inline server/test/test_foo.py]: Missed using the variable here instead of hardcoding :)

File: server/test/test_foo.py [hunk 2]
```diff
@@ -1,70 +1,65 @@
...
-        self.assertEqual(data['example']['string_field'], 'I am a test string')
+        self.assertEqual(data['string_field'], 'I am a test string')
 
-    def test_delete_example(self):
-        create_example_string('example_string_1')
+    def test_delete_foo(self):
+        create_foo_string('foo_string_1')
         with self.client:
-            response = self.client.delete('/api/example_endpoint',
+            response = self.client.delete('/api/foo/1',
```
Comment (fwkendall) [inline server/test/test_foo.py]: Hardcoded id here too :)

File: server/test/test_foo.py [hunk 3]
```diff
@@ -1,70 +1,65 @@
...
+        data = json.loads(response.data.decode())
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(data['records']), 3)
 
-class TestExampleService(BaseTestCase):
-    def test_get_example(self):
-        create_example_string('example_string_1')
-        create_example_string('example_string_2')
-        create_example_string('example_string_3')
+    def test_get_foo(self):
```
Comment (fwkendall) [inline server/test/test_foo.py:18]: Yay! 👍

File: server/manage.py [hunk 4]
```diff
@@ -21,7 +21,7 @@ def create_db():
 # When data need to be seeded this is what we should use
```
Comment (fwkendall) [inline server/manage.py]: I guess we could update this comment now 🤷‍♀

File: server/app/api/blueprints/foo.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+from sqlalchemy import exc
+from app import db
+from app.api.models.foo import Foo
+
+foo_blueprint = Blueprint('foo', __name__)
+
+
+@foo_blueprint.route('/api/foo', methods=['POST'], strict_slashes=False)
+def post():
+
```
Comment (fwkendall) [inline server/app/api/blueprints/foo.py]: Tiny thing, but if you're making other changes anyway could you please add a comment to these indicating they're the default response, or delete the blank line from each method?

File: server/app/api/blueprints/foo.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
+from flask import Blueprint, jsonify, request
+from sqlalchemy import exc
+from app import db
+from app.api.models.foo import Foo
+
+foo_blueprint = Blueprint('foo', __name__)
+
+
+@foo_blueprint.route('/api/foo', methods=['POST'], strict_slashes=False)
```
Comment (fwkendall) [inline server/app/api/blueprints/foo.py:9]: I FORGOT ALL ABOUT THE STRICT SLASHES

Thank you for fixing this 😁 I had to do it during the code challenge too lol

File: server/app/api/blueprints/foo.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+@foo_blueprint.route('/api/foo', methods=['GET'], strict_slashes=False)
+def get_all():
+    response_object = {
+        'records': [foos.to_json() for foos in Foo.query.all()]
+    }
+    return jsonify(response_object), 200
+
+
+@foo_blueprint.route('/api/foo/<int:foo_id>', methods=['GET'], strict_slashes=False)
+def get(foo_id):
```
Comment (fwkendall) [inline server/app/api/blueprints/foo.py:40]: Thank you 🎉 I forgot all about the param/ID fixes too!

File: server/app/api/blueprints/foo.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    record = Foo.query.filter_by(id=foo_id).first()
+    if not record:
+        return jsonify({
+            'errors': [f'No record with id={foo_id} found.']
```
Comment (fwkendall) [inline server/app/api/blueprints/foo.py:44]: Missing error code? We could add a test for this case, too

**Review by fwkendall — COMMENTED** [review]
There's a few small things we should clean up here still :) It looks great so far!

File: server/app/api/blueprints/foo.py (hunk 8 above)
Comment (NWCalvank) [reply server/app/api/blueprints/foo.py:44]: I wondered about this (and most of the things you mentioned in your comments), but I leaned on the side of leaving things "as-is" if they weren't mentioned in the ticket or included in the referenced commits from the code challenge.

So, I can make all these changes and any others if you think of them 👍

Comment (fwkendall) [reply server/app/api/blueprints/foo.py:44]: I totally just forgot these small things lol

I must have fixed them in other commits throughout the code challenge 😓

**Comment by fwkendall** [comment]
Updates look good :)

I know there was something else @gyanesh-mishra wanted to see cleaned up in the model, but I'm not sure what it was. Something about simplifying the code there? Maybe that's a different ticket, or is it pretty small?

**Pull request merged**